docs/UpdatingConfigurations.md

```diff
@@ -0,0 +1,356 @@
+          </executions>
+      </plugin>
+…
+  </plugins>
+</build>
+```
+
+3. Build the component using the `mvn clean install` command and if everything is done correctly, the configuration document file(`<config-namespace-value>.yaml`) will create automatically inside the `<CLASS_OUTPUT_DIRECTORY>/config-docs` directory.
+
+4. Add the Maven dependency plugin to the feature's POM file to copy the configuration document file to the feature. When the feature builds, the configuration document file (`<config-namespace-value>.yaml`) will be copied to the config-docs directory.  
+ 
+ ```xml
+ <plugins>
+…
+  <plugin>
+      <groupId>org.apache.maven.plugins</groupId>
+      <artifactId>maven-dependency-plugin</artifactId>
+      <executions>
+…
+          <execution>
+              <id>unpack</id>
+              <phase>package</phase>
+              <goals>
+                  <goal>unpack</goal>
+              </goals>
+              <configuration>
+                  <artifactItems>
+                      <artifactItem>
+                          <groupId><-- component-group-id --></groupId>
+                          <artifactId><-- component-artifact-id --></artifactId>
+                          <version><-- component-version --></version>
+                          <type>bundle</type>
+                          <overWrite>true</overWrite>
+                          <outputDirectory>${project.build.directory}/docs</outputDirectory>
+                          <includes>config-docs/**</includes>
+                      </artifactItem>
+                  </artifactItems>
+              </configuration>
+          </execution>
+…
+      </executions>
+  </plugin>
+…
+</plugins> 
+```
+
+5. And add resource as shown below.
+
+  ```xml  
+<build>
+…
+<resources>
+…
+  <resource>
+      <directory>${project.build.directory}/docs/</directory>
+  </resource>
+…
+</resources>
+…
+</build> 
+```
+
+Shown below is a sample Configuration Doc file. This is generated for the `CarbonConfiguration` class. If you need to override the default configuations, you need to copy this configs to the `deployment.yaml` file in your server.
+
+```yaml
+################################################################################
+#   Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved
+#
+#   Licensed under the Apache License, Version 2.0 (the \"License\");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an \"AS IS\" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+################################################################################
```

Comment:
As I feel the lines with long hashes are highlighting too much. It might be better if we can remove them. Please see following YAML file: https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/flexible/helloworld/src/main/appengine/app.yaml